src/dbnode/persist/fs/cross_block_reader.go

```diff
@@ -0,0 +1,168 @@
+package fs
+
+import (
+	"bytes"
+	"container/heap"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/m3db/m3/src/x/checked"
+	xerrors "github.com/m3db/m3/src/x/errors"
+	"github.com/m3db/m3/src/x/ident"
+)
+
+var (
+	errReaderNotOrderedByIndex = errors.New("CrossBlockReader can only use DataFileSetReaders ordered by index")
+	_ CrossBlockReader = (*crossBlockReader)(nil)
+	_ heap.Interface   = (*minHeap)(nil)
+)
+
+type crossBlockReader struct {
+	dataFileSetReaders []DataFileSetReader
+	initialized        bool
+	minHeap            minHeap
+	err                error
+}
+
+type minHeapEntry struct {
```

Comment:
nit: I've generally seen type defs followed by the method definitions/closer to where they are used.

Reply:
👍